File: api/company.py
import json
from urllib.request import Request, urlopen

FINANCIAL_API_KEY = "afb05902415e550cb4a4cccefa62d2a1"  # Replace after rotating
FINANCIAL_BASE = "https://api.financialdata.net/v3"

def handler(request):
    query = {k: v[0] for k, v in request.get_query().items()} if request.get_query() else {}
    symbol = query.get("symbol")
    if not symbol:
        return {"statusCode": 400, "body": json.dumps({"error": "Missing symbol"})}
    
    url = f"{FINANCIAL_BASE}/company/profile/{symbol.upper()}?token={FINANCIAL_API_KEY}"
    try:
        with urlopen(Request(url), timeout=10) as resp:
            data = json.loads(resp.read().decode())
        result = {
            "name": data.get("companyName") or data.get("name"),
            "sector": data.get("sector"),
            "employees": data.get("employees") or data.get("fullTimeEmployees"),
            "marketCap": data.get("marketCap") or data.get("market_cap"),
        }
        return {"statusCode": 200, "body": json.dumps(result), "headers": {"Content-Type": "application/json"}}
    except Exception as e:
        return {"statusCode": 500, "body": json.dumps({"error": str(e)})}

File: api/company.js
export default async function handler(req, res) {
    const { symbol } = req.query;
    if (!symbol) return res.status(400).json({ error: 'Missing symbol' });
    const symbolUpper = symbol.toUpperCase();

    // Try Finnhub first (fast, free)
    if (process.env.FINNHUB_API_KEY) {
        try {
            const url = `https://finnhub.io/api/v1/stock/profile2?symbol=${symbolUpper}&token=${process.env.FINNHUB_API_KEY}`;
            const response = await fetch(url);
            const data = await response.json();
            if (data && data.name) {
                return res.status(200).json({
                    name: data.name,
                    sector: data.finnhubIndustry || data.sector,
                    employees: data.employeeTotal,
                    marketCap: data.marketCapitalization,
                });
            }
        } catch (err) { console.warn('Finnhub profile failed:', err.message); }
    }

    // Fallback to FinancialData.net
    if (process.env.FINANCIAL_API_KEY) {
        try {
            const url = `https://api.financialdata.net/api/v1/companies/profile/${symbolUpper}?token=${process.env.FINANCIAL_API_KEY}`;
            const response = await fetch(url);
            const data = await response.json();
            if (data && (data.companyName || data.name)) {
                return res.status(200).json({
                    name: data.companyName || data.name,
                    sector: data.sector,
                    employees: data.employees || data.fullTimeEmployees,
                    marketCap: data.marketCap || data.market_cap,
                });
            }
        } catch (err) { console.warn('FinancialData.net company failed:', err.message); }
    }

    // Last resort: FMP
    if (process.env.FMP_API_KEY) {
        try {
            const url = `https://financialmodelingprep.com/api/v3/profile/${symbolUpper}?apikey=${process.env.FMP_API_KEY}`;
            const response = await fetch(url);
            const data = await response.json();
            if (data && data[0]) {
                const profile = data[0];
                return res.status(200).json({
                    name: profile.companyName,
                    sector: profile.sector,
                    employees: profile.fullTimeEmployees,
                    marketCap: profile.mktCap,
                });
            }
        } catch (err) { console.warn('FMP company failed:', err.message); }
    }

    return res.status(503).json({ error: 'No company profile available' });
}
